src/player.js

```diff
@@ -1812,8 +1879,8 @@ export default class Player extends FakeEventTarget {
     }
     if (type) {
       const tracks = this._getTracksByType(type);
-      for (let i = 0; i < tracks.length; i++) {
-        tracks[i].active = track.index === i;
+      for (let aTrack of tracks) {
```

Comment:
aTrack?

Reply:
Why to change the previous impl?

Reply:
reverted but modified the condition to
track[i].active = track.index === tracks[i].index

becuase this is dependent on the tracks.length and there might be an index higher than the length (because of the use of external)